pkg/sql/schema_changer.go

```diff
@@ -1562,30 +1562,39 @@ func (sc *SchemaChanger) done(ctx context.Context) error {
 				}
 			}
 
-			// If a primary index swap or any indexes are being dropped clean up any
-			// comments related to it.
+			// If `m` is a primary index swap, which results in creations of a
+			// primary index and possibly new, rewritten secondary indexes, carry over
+			// the comments associated with the old indexes to the new ones.
```

Comment:
small grammatical nit:
```suggestion
			// If `m` is a primary index swap, which results in the creation of a
			// primary index and possibly new, rewritten, secondary indexes, carry over
			// the comments associated with the old indexes to the new ones.
```